kebechet/utils.py

```diff
@@ -55,8 +55,32 @@ def cwd(path: str):
         os.chdir(previous_dir)
 
 
-def _clone_repo_and_set_vals(repo_url, repo_path, **clone_kwargs) -> git.Repo:
-    _LOGGER.info(f"Cloning repository {repo_url} to {repo_path}")
+def _clone_repo_and_set_vals(
+    manager: ManagerBase, repo_path: str, **clone_kwargs
+) -> git.Repo:
+    service_url = manager.service_url
+    slug = manager.slug
+    if service_url.startswith("https://"):
+        service_url = service_url[len("https://") :]
+    elif service_url.startswith("http://"):
+        service_url = service_url[len("http://") :]
+    else:
+        # This is mostly internal error - we require service URL to have protocol explicitly set
+        raise NotImplementedError
+
+    namespace, repository = slug.split("/")
+    if manager.installation:
+        access_token = manager.service.authentication.get_token(namespace, repository)
+        repo_url = f"https://{APP_NAME}:{access_token}@{service_url}/{slug}"
+    else:
+        repo_url = f"git@{service_url}:{slug}.git"
+    masked_repo_url = (
+        repo_url.replace(access_token, "MASKED")
+        if access_token is not None
+        else repo_url
+    )
```

Comment:
It seems `access_token` might be undefined here if we go through the `else` branch in the `if` above, right?

```suggestion
    access_token = None
    if manager.installation:
        access_token = manager.service.authentication.get_token(namespace, repository)
        repo_url = f"https://{APP_NAME}:{access_token}@{service_url}/{slug}"
    else:
        repo_url = f"git@{service_url}:{slug}.git"
    masked_repo_url = (
        repo_url.replace(access_token, "MASKED")
        if access_token is not None
        else repo_url
    )
```